forte/data/multi_pack.py

```diff
@@ -833,40 +851,25 @@ def get(  # type: ignore
         else:
             entry_type_ = entry_type
 
-        entry_iter: Iterator[Entry]
-
-        if not include_sub_type:
-            entry_iter = self.get_entries_of(entry_type_)
-        elif issubclass(entry_type_, MultiPackLink):
-            entry_iter = self.links
-        elif issubclass(entry_type_, MultiPackGroup):
-            entry_iter = self.groups
-        elif issubclass(entry_type_, MultiPackGeneric):
-            entry_iter = self.generics
-        else:
-            raise ValueError(
-                f"The entry type: {entry_type_} is not supported by MultiPack."
-            )
-
-        all_types: Set[Type]
-        if include_sub_type:
-            all_types = self._expand_to_sub_types(entry_type_)
-
         if components is not None:
             if isinstance(components, str):
                 components = [components]
 
-        for e in entry_iter:
-            # Will check for the type matching if sub types are also requested.
-            if include_sub_type and type(e) not in all_types:
-                continue
-
-            # Check for the component.
-            if components is not None:
-                if not self.is_created_by(e, components):
-                    continue
+        try:
+            for entry_data in self._data_store.get(
+                type_name=get_full_module_name(entry_type_),
+                include_sub_type=include_sub_type,
+            ):
+                entry: Entry = self.get_entry(tid=entry_data[TID_INDEX])
```

Comment:
Does this `get_entry` affect the efficiency? We get the data but we only take the `tid` and get the entry data again from data store?